Check/Abstract.php

```diff
@@ -279,4 +279,67 @@ public function getPercentOverride() {
     return $this->percentOverride;
   }
 
+  /**
+   * Returns the values of the valid options for a command.
+   *
+   * This function is used by the codebase report where it needs to get the
+   * value of options for different tools it uses. The function takes an
+   * array of valid options to check and a prefix for those options.
+   *
+   * @param array $options
+   *   An array containing the options to be checked and their default values.
+   * @param string $option_prefix
+   *   Prefix for the options.
+   *
+   * @return array
+   *   An associative array containing the value of the options indexed by
+   *   option name.
+   */
+  public function getOptions(array $options, $option_prefix) {
+    $values = array();
+    foreach ($options as $option => $default) {
+      $value = drush_get_option($option_prefix . $option, $default);
+      if ($value !== NULL) {
+        $values[$option] = $value;
+      }
+    }
+    return $values;
+  }
+
+  /**
+   * Returns an array containing custom code paths or AUDIT_CHECK_SCORE_INFO.
+   *
+   * @return array|int
+   *   An array contaning custom code paths or AUDIT_CHECK_SCORE_INFO if custom
+   *   code paths are not found.
+   */
+  public function getCustomCodePaths() {
+    $custom_code = \Drupal::config('site_audit')->get('custom_code');
+    if ($custom_code == NULL) {
+      $custom_code = drush_get_option('custom_code', '');
+      if (empty($custom_code)) {
+        return SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO;
+      }
+      $custom_code = explode(',', $custom_code);
+    }
+    return $custom_code;
+  }
+
+  /**
+   * Returns the path of the PHPMD executable.
```

Comment:
I like that you're abstracting this, but it's not just for `PHPMD`. Similar to the custom code paths, don't return `AUDIT_CHECK_SCORE_INFO`, return an empty string.